internal/model/args.go

```diff
@@ -19,15 +20,16 @@ type LinkArgs struct {
 }
 
 type Link struct {
-	URL        string         `json:"url"`
-	Header     http.Header    `json:"header"` // needed header (for url) or response header(for data or writer)
-	Data       io.ReadCloser  // return file reader directly
-	Status     int            // status maybe 200 or 206, etc
-	FilePath   *string        // local file, return the filepath
-	Expiration *time.Duration // url expiration time
+	URL             string            `json:"url"`
+	Header          http.Header       `json:"header"` // needed header (for url) or response header(for data or writer)
+	RangeReadCloser RangeReadCloser   // recommended way
```

Comment:
It seems that there is no consideration to return a Reader but does not support Range?

Reply:
yes, that's one of the breaking change. all morden web servers supports http range, and we should take advantage of it. the benfits: 
- copy/moving/streaming with a size check at end, avoid incomplete operations, enable retry on unstable networks
- raipid upload function acorss different drives. e.g Aliyun need a small portion of data now to finish the raipid-uploading
- the encryption/decryption need this to work

a workaround with writer is get all the data and ditch all the data until offset. but this will be terrible both in design and performance. 

I just checked no usage of this linkStreamAlbum and linkStream, can we implement it using RangeReadCloser , ReadSeekCloser, or link_url?

Reply:
you may have misunderstood my meaning, I just asked if the `linkStreamAlbum` and `linkStream` functions in the BaiduPhoto driver support range

Reply:
> Most of them are supported, only a very small number of them are not (not supporting them is equivalent to not being able to multithread downloads, and the experience won't be good enough to be adapted by alist). The ones that don't are easy to fix, either by returning a specific error or by using a cached Reader.

i've rewrote every one that doesnot support range before submit this pr. it's just the new commit u did yesterday that still use "Writer",
1.  I don't see the function(linkStreamAlbum &linkStream) is used anywhere, is this a prepare for some future function?
2.  I did not find any doc on this (album/v1/streaming)? is this a reverse-engineered API ?

Reply:
It's not used yet, just in case I forget about the api. You can just comment out the code.
It should be used for the Other method

Reply:
> It's not used yet, just in case I forget about the api. You can just comment out the code. It should be used for the Other method

thanks for the information, i think it's very clear now:
1. linkStreamAlbum and linkStream will not feed data to Link/List function directly. (this is very important, since i di some very big changed to how Alist deal with Link/List returned &model.Link )
2. i think we can commet it out for now, and in the future when u try to implement m3u8 function, u can choose: 
  - it's small enough to buffer in the memeory, so return the RangeReadCloser/ReadSeekCloser wraping the buffer
  - if no need to use any alist download/serveHttp function, just define a new struct for your internal use with Writer.